src/Service/UserService.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\User;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class UserService
+{
+    private EntityManagerInterface $entityManager;
+    private ValidatorInterface $validator;
+    private UserPasswordHasherInterface $userPasswordHasher;
+    private MailerService $mailerService;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        ValidatorInterface $validator,
+        UserPasswordHasherInterface $userPasswordHasher,
+        MailerService $mailerService
+    ) {
+        $this->entityManager = $entityManager;
+        $this->validator = $validator;
+        $this->userPasswordHasher = $userPasswordHasher;
+        $this->mailerService = $mailerService;
+    }
+
+    public function processUserData(Request $request): array
+    {
+        // data validation using separate validator
+        // update entity for validation unique values, different other conditions
+        // send mail to admin using separate service which should initiate messenger handler
+
+        return $this->saveUserData($request);
+    }
+
+    private function saveUserData(Request $request): array
+    {
+        $user = new User();
+        $user->setName($request->get('name'));
+        $user->setEmail($request->get('email'));
+        $user->setMobileNumber($request->get('mobileNumber'));
+        $user->setEmployeeId($request->get('employeeId'));
+        $user->setPassword(
+            $this->userPasswordHasher->hashPassword($user, $request->get('password'))
+        );
+        $user->setVerificationCode(substr(md5((string) mt_rand(9999, 999999)), 0, 60));
+
+        $validData = $this->validateUserData($user);
```

Comment:
Is it possible for us to create a dedicated validator service for handling data validation?

Reply:
I have already added a validator named UserValidator in another PR.